entrypoint.sh

```diff
@@ -1,236 +1,227 @@
+  sed -i '/^addnode/d' "${HOME}/.zen/zen.conf" "${HOME}/.zen/testnet3/zen.conf"
+  mapfile -t addnode <<< "$(tr ',' '\n' <<< "${ADDNODE}")"
+  echo -e "Adding addnode= for: ${addnode[*]}\n"
+  for node in "${addnode[@]}"; do
+    echo "addnode=${node}" | tee -a "${HOME}/.zen/zen.conf" >> "${HOME}/.zen/testnet3/zen.conf"
+  done
 fi
 
 # TLS_KEY_PATH path to SSL private key inside of the container
 # Example: TLS_KEY_PATH="/home/user/.zen/ssl.key"
 
 # Set tlskeypath= in zen.conf
-if [[ ! -z "$TLS_KEY_PATH" ]]; then
-    sed -i '/^tlskeypath/d' $HOME/.zen/zen.conf $HOME/.zen/testnet3/zen.conf
-    echo "tlskeypath=${TLS_KEY_PATH}" | tee -a $HOME/.zen/zen.conf >> $HOME/.zen/testnet3/zen.conf
+if [ -n "${TLS_KEY_PATH:-}" ]; then
+  [ -f "${TLS_KEY_PATH}" ] || { echo "Error: no TLS key file found at ${TLS_KEY_PATH}"; sleep 5; exit 1; }
+  sed -i '/^tlskeypath/d' "${HOME}/.zen/zen.conf" "${HOME}/.zen/testnet3/zen.conf"
+  echo "tlskeypath=${TLS_KEY_PATH}" | tee -a "${HOME}/.zen/zen.conf" >> "${HOME}/.zen/testnet3/zen.conf"
 fi
 
 # TLS_CERT_PATH path to SSL certificate inside of the container
 # Example: TLS_CERT_PATH="/home/user/.zen/ssl.crt"
 
 # Set tlscertpath= in zen.conf
-if [[ ! -z "$TLS_CERT_PATH" ]]; then
-    sed -i '/^tlscertpath/d' $HOME/.zen/zen.conf $HOME/.zen/testnet3/zen.conf
-    echo "tlscertpath=${TLS_CERT_PATH}" | tee -a $HOME/.zen/zen.conf >> $HOME/.zen/testnet3/zen.conf
+if [ -n "${TLS_CERT_PATH:-}" ]; then
+  [ -f "${TLS_CERT_PATH}" ] || { echo "Error: no TLS cert file found at ${TLS_CERT_PATH}"; sleep 5; exit 1; }
+  sed -i '/^tlscertpath/d' "${HOME}/.zen/zen.conf" "${HOME}/.zen/testnet3/zen.conf"
+  echo "tlscertpath=${TLS_CERT_PATH}" | tee -a "${HOME}/.zen/zen.conf" >> "${HOME}/.zen/testnet3/zen.conf"
 fi
 
 # Fix ownership of the created files/folders
-chown -R $LOCAL_UID:$LOCAL_GID $HOME /mnt/zen /mnt/zcash-params
+find "${mountpoints[@]}" -writable -print0 | xargs -0 -I{} -P64 -n1 chown -f "${CURRENT_UID}":"${CURRENT_GID}" "{}"
 
 # CUSTOM_SCRIPT, execute user provided script before starting zend, e.g. to backup wallets
-if [[ ! -z "$CUSTOM_SCRIPT" ]]; then
-    chmod +x "${CUSTOM_SCRIPT}"
-    echo "Running custom script: ${CUSTOM_SCRIPT}"
-    bash -c "${CUSTOM_SCRIPT}"
+if [ -n "${CUSTOM_SCRIPT:-}" ]; then
+  chmod +x "${CUSTOM_SCRIPT}"
+  echo "Running custom script: ${CUSTOM_SCRIPT}"
+  bash -c "${CUSTOM_SCRIPT}"
 fi
 
-if [ ! "$USER_ID" == "0"  ]; then
-    if [[ "$1" == zend ]]; then
-        /usr/local/bin/gosu user zen-fetch-params
-        exec /usr/local/bin/gosu user /bin/bash -c "$@ $OPTS"
-    fi
-    exec /usr/local/bin/gosu user "$@"
-else
-    if [[ "$1" == zend ]]; then
-        zen-fetch-params
-        exec /bin/bash -c "$@ $OPTS"
-    fi
-    exec "$@"
-fi
+# convert $OPTS into array
+mapfile -t OPTS < <(sed 's/  */ /g; s/ $//g' <<< "${OPTS}" | tr ' ' '\n')
 
+gosu_cmd=""
+[ "${CURRENT_UID}" -ne 0 ] && gosu_cmd="/usr/local/bin/gosu user"
+if [ "$1" = "zend" ]; then
+  $gosu_cmd zen-fetch-params
```

Comment:
Should this be "${gosu_cmd}" as well?

Reply:
In this case we need `/usr/local/bin/gosu` and  `user` to be interpreted as separate commands. That only works if we word split without double quotes.